core/components/scripts/scripts.tsx

```diff
@@ -0,0 +1,121 @@
+  location: 'head' | 'footer';
+  strategy: 'afterInteractive' | 'lazyOnload';
+}
+
+interface ScriptsProps {
+  scripts: ScriptsData | null;
+}
+
+interface ScriptProps {
+  src?: string;
+  dangerouslySetInnerHTML?: {
+    __html: string;
+  };
+  strategy: 'afterInteractive' | 'lazyOnload';
+  integrity?: string;
+  id?: string;
+  crossorigin?: string;
+}
+
+/**
+ * Shared component that renders BigCommerce scripts using Next.js Script component
+ * @param {ScriptRendererProps} props - The props for the script renderer
+ * @param {ScriptsData | null} props.scripts - The scripts data from GraphQL
+ * @param {'head' | 'footer'} props.location - Where to render the scripts
+ * @param {'afterInteractive' | 'lazyOnload'} props.strategy - The loading strategy
+ * @returns {JSX.Element | null} JSX element containing the rendered scripts
+ */
+const ScriptRenderer = ({ scripts, location, strategy }: ScriptRendererProps) => {
+  if (!scripts?.edges) return null;
+
+  const scriptNodes = removeEdgesAndNodes(scripts);
+  const locationScripts = scriptNodes.filter(
+    (script) => script.location.toLowerCase() === location,
+  );
+  const filteredScripts = locationScripts.filter(
+    // only render scripts that are visible on all pages or storefront
+    (script) => script.visibility === 'ALL_PAGES' || script.visibility === 'STOREFRONT',
+  );
+
+  return (
+    <>
+      {filteredScripts.map((script) => {
+        const scriptProps: ScriptProps = {
+          strategy,
+        };
+
+        // Handle external scripts (SrcScript)
+        if (script.__typename === 'SrcScript' && script.src) {
+          scriptProps.src = script.src;
+
+          // Add integrity hashes if provided
+          if (script.integrityHashes.length > 0) {
+            scriptProps.integrity = script.integrityHashes.map((hashObj) => hashObj.hash).join(' ');
+            scriptProps.crossorigin = 'anonymous';
+          }
+        }
+
+        // Handle inline scripts (InlineScript)
+        if (script.__typename === 'InlineScript' && script.scriptTag) {
+          const scriptMatch = /<script[^>]*>([\s\S]*?)<\/script>/i.exec(script.scriptTag);
+
+          if (scriptMatch?.[1]) {
+            scriptProps.dangerouslySetInnerHTML = {
+              __html: scriptMatch[1],
+            };
+          }
+        }
+
+        scriptProps.id = `bc-script-${script.entityId}`;
+
+        return <Script key={scriptProps.id} {...scriptProps} />;
+      })}
+    </>
+  );
+};
+
+/**
+ * Renders scripts that should be placed in the <head> tag
+ * Uses afterInteractive strategy for critical functionality
+ * @param {ScriptsProps} props - The props containing scripts data
+ * @param {ScriptsData | null} props.scripts - The scripts data from GraphQL
+ * @returns {JSX.Element | null} JSX element containing header scripts
+ */
+export const HeaderScripts = ({ scripts }: ScriptsProps) => {
+  return <ScriptRenderer location="head" scripts={scripts} strategy="afterInteractive" />;
+};
```

Comment:
I think this might not be the right abstraction. Can we just utilize the server to filter the scripts and just have one component we utilize: `ScriptRenderer`? See https://github.com/bigcommerce/catalyst/pull/2477/files#r2205218637